xbmc/cores/RetroPlayer/rendering/VideoRenderers/RPBaseRenderer.cpp

```diff
@@ -129,8 +134,10 @@ void CRPBaseRenderer::ManageRenderArea(const IRenderBuffer& renderBuffer)
   const unsigned int sourceRotationDegCCW = renderBuffer.GetRotation();
   const float sourceAspectRatio =
       static_cast<float>(sourceWidth) / static_cast<float>(sourceHeight);
-
+  const SCALINGMETHOD scaleMode = m_renderSettings.VideoSettings().GetScalingMethod();
   const STRETCHMODE stretchMode = m_renderSettings.VideoSettings().GetRenderStretchMode();
+  const bool bhw_filter =
+	  CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(SETTING_VIDEOSCREEN_HW_SCALING_FILTER);
```

Comment:
we should check for nullptr when getting settings

Reply:
Yes, I will do the changes accordingly.